### ISSUE TITLE
Agent manager is crashing when I try to parallelize a simple piece of code

### ISSUE THREAD

**Issue by riccardodv** [rlberry-py/rlberry#116]
When I run the following code with AgentManager and parallelization 

```python
import numpy as np
from rlberry.manager import AgentManager
from rlberry.envs.benchmarks.ball_exploration import PBall2D
from rlberry.agents.torch.a2c.a2c import A2CAgent

env_ctor = PBall2D
env_kwargs = dict()
env = env_ctor(**env_kwargs)

agent = AgentManager(A2CAgent, (env_ctor, env_kwargs), fit_budget=100, n_fit=3, max_workers=2, parallelization="process")
agent.fit()
```

results in the following error

```
RuntimeError: 
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.
```

Constructing an agent with AgentManager and adding `mp_context = fork` seems to solve the problem but still it seems a problem.

**Comment by TimotheeMathieu** [comment]
Did you try to execute this code in a notebook ? The default `mp_context = spawn` do not work in a notebook, the parallelization gets messed up and you have to use fork.

**Comment by TimotheeMathieu** [comment]
To follow up, here is a working script :

```
import numpy as np
from rlberry.manager import AgentManager
from rlberry.envs.benchmarks.ball_exploration import PBall2D
from rlberry.agents.torch.a2c.a2c import A2CAgent

env_ctor = PBall2D
env_kwargs = dict()
env = env_ctor(**env_kwargs)

agent = AgentManager(A2CAgent, (env_ctor, env_kwargs), fit_budget=100, n_fit=3, max_workers=2, parallelization="process")
if __name__ == '__main__':
    agent.fit()
```
In short, the explanation is that if you use `paralellization="process"` and `mp_context=spawn` then the parallel computing code must be "protected" behing a `if __name__ == '__main__':`.

* `paralellization="process"`  is to have real paralellization, without this it is not really parallel except if you do a lot of numpy in each fit
* From what I understood, `mp_context=spawn` is mainly because `mp_context=fork` do not work with jax

**Comment by TimotheeMathieu** [comment]
Solved by PR#120, I close.
